rholang/src/main/scala/coop/rchain/rholang/interpreter/sort.scala

```diff
@@ -0,0 +1,311 @@
+      case ee : EEq =>
+        val (sortedPar1, sortedPar2) = sortBinaryOperation(ee.p1, ee.p2)
+        ScoredTerm(EEq(sortedPar1.term, sortedPar2.term),
+          Node(Seq(Leaf(Score.EEQ), sortedPar1.score, sortedPar2.score)))
+      case en : ENeq =>
+        val (sortedPar1, sortedPar2) = sortBinaryOperation(en.p1, en.p2)
+        ScoredTerm(ENeq(sortedPar1.term, sortedPar2.term),
+          Node(Seq(Leaf(Score.ENEQ), sortedPar1.score, sortedPar2.score)))
+      case en: ENot =>
+        val sortedPar = ParSortMatcher.sortMatch(en.p)
+        ScoredTerm(ENot(sortedPar.term),
+          Node(Seq(Leaf(Score.ENOT), sortedPar.score)))
+      case ea : EAnd =>
+        val (sortedPar1, sortedPar2) = sortBinaryOperation(ea.p1, ea.p2)
+        ScoredTerm(EAnd(sortedPar1.term, sortedPar2.term),
+          Node(Seq(Leaf(Score.EAND), sortedPar1.score, sortedPar2.score)))
+      case eo : EOr =>
+        val (sortedPar1, sortedPar2) = sortBinaryOperation(eo.p1, eo.p2)
+        ScoredTerm(EOr(sortedPar1.term, sortedPar2.term),
+          Node(Seq(Leaf(Score.EOR), sortedPar1.score, sortedPar2.score)))
+    }
+  }
+}
+
+object VarSortMatcher {
+  def sortMatch(v : Var) : ScoredTerm[Var] = {
+    v match {
+      case bv : BoundVar => ScoredTerm(bv, Leaves(Score.BOUND_VAR, bv.level))
+      case wc : WildCard => ScoredTerm(wc, Leaf(Score.WILDCARD))
+      case fv : FreeVar => ScoredTerm(fv, Leaves(Score.FREE_VAR, fv.level))
+    }
+  }
+}
+
+object ChannelSortMatcher {
+  def sortMatch(c: Channel) : ScoredTerm[Channel] = {
+    c match {
+      case cq : Quote =>
+        val sortedPar = ParSortMatcher.sortMatch(cq.p)
+        ScoredTerm(Quote(sortedPar.term), Node(Seq(Leaf(Score.QUOTE), sortedPar.score)))
+      case cv : ChanVar =>
+        val sortedVar = VarSortMatcher.sortMatch(cv.cvar)
+        ScoredTerm(ChanVar(sortedVar.term), Node(Seq(Leaf(Score.CHAN_VAR), sortedVar.score)))
+    }
+  }
+}
+
+object SendSortMatcher {
+  def sortMatch(s: Send) : ScoredTerm[Send] = {
+    val sortedChan = ChannelSortMatcher.sortMatch(s.chan)
+    val sortedData = s.data.map(d => ParSortMatcher.sortMatch(d))
+    val sortedSend = Send(chan = sortedChan.term, data = sortedData.map(_.term))
+    val sendScore = Node(Seq(Leaf(Score.SEND)) ++ Seq(sortedChan.score) ++ sortedData.map(_.score))
+    ScoredTerm(sortedSend, sendScore)
+  }
+}
+
+// TODO: Sort the binds
+object ReceiveSortMatcher {
+  def sortBind(bind: (List[Channel], Channel)): ScoredTerm[Tuple2[List[Channel], Channel]] = {
+    val (patterns, channel) = bind
+    val sortedPatterns = patterns.map(channel => ChannelSortMatcher.sortMatch(channel))
+    val sortedChannel = ChannelSortMatcher.sortMatch(channel)
+    ScoredTerm((sortedPatterns.map(_.term), sortedChannel.term),
+                Node(Seq(sortedChannel.score) ++ sortedPatterns.map(_.score)))
+  }
+
+  def sortMatch(r: Receive) : ScoredTerm[Receive] = {
+    val sortedBinds = r.binds.map(bind => sortBind(bind))
```

Comment:
Sorting the binds should be as simple as adding .sorted to this line.

Reply:
Per our conversation, this function should not sort the binds.

Instead, write a helper method that can sort a list of (Channel, List[Channel], DebruijnLevelMap) triples

Reply:
Done